tests/simple/simple.cpp

```diff
@@ -0,0 +1,206 @@
+        pc++;
+        if (Specialized) {
+            weval::update_context(pc);
+        }
+        switch (inst->opcode) {
+            case PushConst:
+                if (sp + 1 > OPSTACK_SIZE) {
+                    return false;
+                }
+                opstack[sp++] = inst->imm;
+                break;
+            case Drop:
+                if (sp == 0) {
+                    return false;
+                }
+                sp--;
+                break;
+            case Dup:
+                if (sp + 1 > OPSTACK_SIZE) {
+                    return false;
+                }
+                if (sp == 0) {
+                    return false;
+                }
+                opstack[sp] = opstack[sp - 1];
+                sp++;
+                break;
+            case GetLocal:
+                if (sp + 1 > OPSTACK_SIZE) {
+                    return false;
+                }
+                if (inst->imm >= LOCAL_SIZE) {
+                    return false;
+                }
+                opstack[sp++] = locals[inst->imm];
+                break;
+            case SetLocal:
+                if (sp == 0) {
+                    return false;
+                }
+                if (inst->imm >= LOCAL_SIZE) {
+                    return false;
+                }
+                locals[inst->imm] = opstack[--sp];
+                break;
+            case Add:
+                if (sp < 2) {
+                    return false;
+                }
+                opstack[sp - 2] += opstack[sp - 1];
+                sp--;
+                break;
+            case Sub:
+                if (sp < 2) {
+                    return false;
+                }
+                opstack[sp - 2] -= opstack[sp - 1];
+                sp--;
+                break;
+            case Print:
+                if (sp == 0) {
+                    return false;
+                }
+                printf("%u\n", opstack[--sp]);
+                break;
+            case Goto:
+                if (inst->imm >= ninsts) {
+                    return false;
+                }
+                pc = inst->imm;
+                if (Specialized) {
+                    weval::update_context(pc);
+                }
+                break;
+            case GotoIf:
+                if (sp == 0) {
+                    return false;
+                }
+                if (inst->imm >= ninsts) {
+                    return false;
+                }
+                sp--;
+                if (opstack[sp] != 0) {
+                    pc = inst->imm;
+                    if (Specialized) {
+                        weval::update_context(pc);
+                    }
+                    continue;
+                }
+                break;
+            case Exit:
+                goto out;
+        }
+    }
+out:
+    if (Specialized) {
+        weval::pop_context();
+    }
+
+    printf("Exiting after %d steps at PC %d.\n", steps, pc);
+    return true;
```

Comment:
It'd be good to check somewhere that we actually get here and this returns `true`. Does `make` error out if the `wasmtime` CLI invocation fails with a non-zero exit status? Also see below about the results potentially not being propagated.

Reply:
Good point! I've switched it around to return the number of steps, and the top-level invocation checks that we ran the expected number of steps for the baked-in bytecode.

Reply:
(And in writing this I've verified that the `make` invocation does indeed fail if the count is wrong!)